### ISSUE TITLE
highlighting not triggered on command queue pop

### ISSUE THREAD

**Issue by sunaku** [zsh-users/zsh-syntax-highlighting#40]
1. Type in a command
2. Press `Alt-q` to queue it
3. Press `Enter` to pop the queued command
4. Observe that popped command is not syntax highlighted

**Comment by danielshahaf** [comment]
The command is [popped from `bufstack`][1] in `zleread`, which [clears all highlighting as the last thing before returning][2]. When `zleread` returns, control leaves the line editor and moves to `inputline()` in the main shell.  I suspect this will need to be fixed in the C code.

```
#0  zleread (lp=0x6dcd80 <prompt>, rp=0x6dcdb8 <rprompt>, flags=3, context=0, init=0x7ffff5e3fab8 "zle-line-init", 
    finish=0x7ffff5e3faa8 "zle-line-finish") at zle_main.c:1201
#1  0x00007ffff5e1bb41 in zle_main_entry (cmd=1, ap=0x7fffffffe2a0) at zle_main.c:1931
#2  0x000000000044e5ae in zleentry (cmd=1) at init.c:1523
#3  0x000000000044f46c in inputline () at input.c:293
```

[1]: https://github.com/zsh-users/zsh/blob/377e2400b7b6fc21ff333791461d615538b18cb9/Src/Zle/zle_main.c#L1201
[2]: https://github.com/zsh-users/zsh/blob/377e2400b7b6fc21ff333791461d615538b18cb9/Src/Zle/zle_main.c#L1295

**Comment by danielshahaf** [comment]
Patch: [workers/39432](www.zsh.org/cgi-bin/mla/redirect?WORKERNUMBER=39432)

**Comment by danielshahaf** [comment]
Different patch committed upstream: zsh-users/zsh@5cf2ffb327faf84b73beb812ff2f7e9bf08e7ecf (workers/39452)

The fix requires both new zsh (current master, 5.3-to-be) and new z-sy-h (presumably due to #245).

**Comment by danielshahaf** [comment]
Will be fixed by #245.
